DC monitor: sample job failure

Sample job "Copy employees to customer table" fails with the error 
```

10:14:11 - Job execution BEGIN
10:14:11 - Starting processing of EMPLOYEES
 - Query: SELECT "EMPLOYEES"."EMPLOYEENUMBER", "EMPLOYEES"."LASTNAME", "EMPLOYEES"."FIRSTNAME", "EMPLOYEES"."EXTENSION", "EMPLOYEES"."EMAIL", "EMPLOYEES"."OFFICECODE", "EMPLOYEES"."REPORTSTO", "EMPLOYEES"."JOBTITLE" FROM PUBLIC."EMPLOYEES"
 - Expected row count: 23
10:14:11 - Progress of EMPLOYEES: 2 rows processed
10:14:11 - 10 inserts executed (orderdb - CUSTOMERS (18 columns))
10:14:11 - 10 inserts executed (orderdb - CUSTOMERS (18 columns))
10:14:11 - Job execution FAILURE
 - Could not execute batch: org.hsqldb.jdbc.jdbcPreparedStatement@1903157[sql=[INSERT INTO PUBLIC."CUSTOMERS" (CUSTOMERNUMBER,CUSTOMERNAME,CONTACTLASTNAME,CONTACTFIRSTNAME,PHONE,ADDRESSLINE1,CITY,COUNTRY,SALESREPEMPLOYEENUMBER) VALUES (?,?,?,?,?,?,?,?,?)], parameters=[[5010], [My company], [Patterson], [Steve], [My phone no], [My address], [My city], [My country], [1216]]]: failed batch (MetaModelException)
 - Failure component: ImmutableAnalyzerJob[name=null,analyzer=Insert into table]
 - Failure input data: TransformedInputRow[values={TransformedInputColumn[id=trans-0001-0002,name=Generated ID]=5011},delegate=MetaModelInputRow[Row[values=[1286, Tseng, Foon Yue, x2248, ftseng@classicmodelcars.com, 3, 1143, Sales Rep]]]]
 - Exception stacktrace of failure condition:
org.apache.metamodel.MetaModelException: Could not execute batch: org.hsqldb.jdbc.jdbcPreparedStatement@1903157[sql=[INSERT INTO PUBLIC."CUSTOMERS" (CUSTOMERNUMBER,CUSTOMERNAME,CONTACTLASTNAME,CONTACTFIRSTNAME,PHONE,ADDRESSLINE1,CITY,COUNTRY,SALESREPEMPLOYEENUMBER) VALUES (?,?,?,?,?,?,?,?,?)], parameters=[[5010], [My company], [Patterson], [Steve], [My phone no], [My address], [My city], [My country], [1216]]]: failed batch
	at org.apache.metamodel.jdbc.JdbcUtils.wrapException(JdbcUtils.java:70)
	at org.apache.metamodel.jdbc.JdbcBatchUpdateCallback.closePreparedStatement(JdbcBatchUpdateCallback.java:64)
	at org.apache.metamodel.jdbc.JdbcUpdateCallback.close(JdbcUpdateCallback.java:80)
	at org.apache.metamodel.jdbc.JdbcDataContext.executeUpdate(JdbcDataContext.java:833)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:414)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:76)
	at org.datacleaner.util.WriteBuffer.flushBuffer(WriteBuffer.java:84)
	at org.datacleaner.util.WriteBuffer.addToBuffer(WriteBuffer.java:60)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:377)
	at org.datacleaner.job.runner.AnalyzerConsumer.consumeInternal(AnalyzerConsumer.java:71)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.processNext(ConsumeRowHandlerDelegate.java:90)
	at org.datacleaner.job.runner.TransformerConsumer.consumeInternal(TransformerConsumer.java:113)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandler.consumeRow(ConsumeRowHandler.java:146)
	at org.datacleaner.job.tasks.ConsumeRowTask.execute(ConsumeRowTask.java:51)
	at org.datacleaner.job.concurrent.TaskRunnable.run(TaskRunnable.java:61)
	at org.datacleaner.job.concurrent.MultiThreadedTaskRunner.assistExecution(MultiThreadedTaskRunner.java:129)
	at org.datacleaner.monitor.configuration.SharedTaskRunner.assistExecution(SharedTaskRunner.java:64)
	at org.datacleaner.job.runner.ErrorAwareTaskRunnerWrapper.assistExecution(ErrorAwareTaskRunnerWrapper.java:95)
	at org.datacleaner.job.runner.RowConsumerTaskListener.awaitTasks(RowConsumerTaskListener.java:73)
	at org.datacleaner.job.runner.SourceTableRowProcessingPublisher.processRowsInternal(SourceTableRowProcessingPublisher.java:244)
	at org.datacleaner.job.runner.AbstractRowProcessingPublisher.processRows(AbstractRowProcessingPublisher.java:208)
	at org.datacleaner.job.tasks.RunRowProcessingPublisherTask.execute(RunRowProcessingPublisherTask.java:43)
	at org.datacleaner.job.concurrent.TaskRunnable.run(TaskRunnable.java:61)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.sql.BatchUpdateException: failed batch
	at org.hsqldb.jdbc.jdbcStatement.executeBatch(Unknown Source)
	at org.hsqldb.jdbc.jdbcPreparedStatement.executeBatch(Unknown Source)
	at org.apache.commons.dbcp.DelegatingStatement.executeBatch(DelegatingStatement.java:297)
	at org.apache.commons.dbcp.DelegatingStatement.executeBatch(DelegatingStatement.java:297)
	at org.apache.metamodel.jdbc.JdbcBatchUpdateCallback.closePreparedStatement(JdbcBatchUpdateCallback.java:45)
	... 28 more

Check the server logs for more details, warnings and debug information.
10:14:11 - 
 - Additional exception stacktrace:
org.apache.metamodel.MetaModelException: Could not execute batch: org.hsqldb.jdbc.jdbcPreparedStatement@27d9d5[sql=[INSERT INTO PUBLIC."CUSTOMERS" (CUSTOMERNUMBER,CUSTOMERNAME,CONTACTLASTNAME,CONTACTFIRSTNAME,PHONE,ADDRESSLINE1,CITY,COUNTRY,SALESREPEMPLOYEENUMBER) VALUES (?,?,?,?,?,?,?,?,?)], parameters=[[5023], [My company], [Gerard], [Martin], [My phone no], [My address], [My city], [My country], [1702]]]: failed batch
	at org.apache.metamodel.jdbc.JdbcUtils.wrapException(JdbcUtils.java:70)
	at org.apache.metamodel.jdbc.JdbcBatchUpdateCallback.closePreparedStatement(JdbcBatchUpdateCallback.java:64)
	at org.apache.metamodel.jdbc.JdbcUpdateCallback.close(JdbcUpdateCallback.java:80)
	at org.apache.metamodel.jdbc.JdbcDataContext.executeUpdate(JdbcDataContext.java:833)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:414)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:76)
	at org.datacleaner.util.WriteBuffer.flushBuffer(WriteBuffer.java:84)
	at org.datacleaner.util.WriteBuffer.addToBuffer(WriteBuffer.java:60)
	at org.datacleaner.beans.writers.InsertIntoTableAnalyzer.run(InsertIntoTableAnalyzer.java:377)
	at org.datacleaner.job.runner.AnalyzerConsumer.consumeInternal(AnalyzerConsumer.java:71)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.processNext(ConsumeRowHandlerDelegate.java:90)
	at org.datacleaner.job.runner.TransformerConsumer.consumeInternal(TransformerConsumer.java:113)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandler.consumeRow(ConsumeRowHandler.java:146)
	at org.datacleaner.job.tasks.ConsumeRowTask.execute(ConsumeRowTask.java:51)
	at org.datacleaner.job.concurrent.TaskRunnable.run(TaskRunnable.java:61)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.sql.BatchUpdateException: failed batch
	at org.hsqldb.jdbc.jdbcStatement.executeBatch(Unknown Source)
	at org.hsqldb.jdbc.jdbcPreparedStatement.executeBatch(Unknown Source)
	at org.apache.commons.dbcp.DelegatingStatement.executeBatch(DelegatingStatement.java:297)
	at org.apache.commons.dbcp.DelegatingStatement.executeBatch(DelegatingStatement.java:297)
	at org.apache.metamodel.jdbc.JdbcBatchUpdateCallback.closePreparedStatement(JdbcBatchUpdateCallback.java:45)
	... 20 more

```

--- a/monitor/ui/src/main/webapp/repository/demo/jobs/Copy employees to customer table.analysis.xml
+++ b/monitor/ui/src/main/webapp/repository/demo/jobs/Copy employees to customer table.analysis.xml
@@ -28,35 +28,29 @@
     </transformation>
     <analysis>
         <analyzer>
-            <descriptor ref="Insert into table"/>
+            <descriptor ref="Create CSV file"/>
+            <metadata-properties>
+                <property name="CoordinatesY">205</property>
+                <property name="CoordinatesX">354</property>
+            </metadata-properties>
             <properties>
-                <property name="Buffer size" value="TINY"/>
-                <property name="Column names" value="[SALESREPEMPLOYEENUMBER,CONTACTLASTNAME,CONTACTFIRSTNAME,CUSTOMERNUMBER,CUSTOMERNAME,PHONE,ADDRESSLINE1,CITY,COUNTRY]"/>
-                <property name="Datastore" value="orderdb"/>
-                <property name="Error log file location" value="C:/Users/kasper/AppData/Local/Temp"/>
-                <property name="How to handle insertion errors?" value="STOP_JOB"/>
-                <property name="Schema name" value="PUBLIC"/>
-                <property name="Table name" value="CUSTOMERS"/>
-                <property name="Truncate table" value="false"/>
+                <property name="File" value="file://employees.csv"/>
+                <property name="Separator char" value="&amp;#44;"/>
+                <property name="Quote char" value="&amp;quot;"/>
+                <property name="Escape char" value="\"/>
+                <property name="Include header" value="true"/>
+                <property name="Encoding" value="UTF-8"/>
+                <property name="Fields" value="[EMPLOYEENUMBER,LASTNAME,FIRSTNAME,EXTENSION,EMAIL,OFFICECODE,REPORTSTO,JOBTITLE]"/>
+                <property name="Overwrite file if exists" value="true"/>
             </properties>
-            <input ref="col_8" name="Additional error log values"/>
-            <input ref="col_0" name="Additional error log values"/>
-            <input ref="col_1" name="Additional error log values"/>
-            <input ref="col_2" name="Additional error log values"/>
-            <input ref="col_3" name="Additional error log values"/>
-            <input ref="col_4" name="Additional error log values"/>
-            <input ref="col_5" name="Additional error log values"/>
-            <input ref="col_6" name="Additional error log values"/>
-            <input ref="col_7" name="Additional error log values"/>
-            <input ref="col_0" name="Values"/>
-            <input ref="col_1" name="Values"/>
-            <input ref="col_2" name="Values"/>
-            <input ref="col_8" name="Values"/>
-            <input value="My company" name="Values"/>
-            <input value="My phone no" name="Values"/>
-            <input value="My address" name="Values"/>
-            <input value="My city" name="Values"/>
-            <input value="My country" name="Values"/>
+            <input ref="col_0" name="Columns"/>
+            <input ref="col_1" name="Columns"/>
+            <input ref="col_2" name="Columns"/>
+            <input ref="col_3" name="Columns"/>
+            <input ref="col_4" name="Columns"/>
+            <input ref="col_5" name="Columns"/>
+            <input ref="col_6" name="Columns"/>
+            <input ref="col_7" name="Columns"/>
         </analyzer>
     </analysis>
 </job>

